### DISCUSSION TITLE
Any factors / edge cases to be aware of when switching Publii themes?

### DISCUSSION THREAD

**Discussion by jjets718** [GetPublii/Publii#1656] [Help]
Hi there!

In the next week, I'm planning to switch my Publii theme from Square to a different one—likely to Persona (https://marketplace.getpublii.com/themes/persona/).

Is there anything I should keep in mind when making this switch? Will all of my existing settings transfer over from Square to Persona when I make the change? Are there any edge cases I should be aware of? Thank you!

**Comment by Backspaze** [comment]
I'm certainly no expert and don't have a checklist or similar for this scenario, but one thing you could do is to [clone your site](https://getpublii.com/docs/creating-your-website.html#duplicatingawebsite) and try the new theme on the clone first to see how it works.

**Reply by jjets718** [reply]
That's a great idea. I'm going to do that. Thank you!

**Comment by candidexmedia** [comment]
Basically: Your Site Settings will remain, and your Theme settings will not:

![Publii_Menu](https://github.com/user-attachments/assets/c55de858-aee8-4d53-8a4b-60b9193c4a5c)

Each theme has its own set of settings, and those options don't necessarily translate over (or make sense) for a different theme. That said, if you switch back (without uninstalling that theme for that site), your Theme settings will stay intact. 

The best way to understand this is to open up your [site folder](https://getpublii.com/docs/publii-global-settings.html#fileslocation) in your file browser or code editor (I use VS Codium), and look at the theme folder. In the screenshot below, you can see that my site has several themes installed, each with their own folder containing the settings, stylesheets, partials, etc.:

![image](https://github.com/user-attachments/assets/af525796-a2fe-4118-b8a9-baa56b7080af)

If you added some [Custom CSS](https://getpublii.com/docs/tools.html#customcss) via the Publii GUI, it will remain. That said, the override style rules might not make sense with the new theme.

![image](https://github.com/user-attachments/assets/99fbdf9c-6487-49eb-b15b-358ff71929de)

Your posts, authors, and tags will remain the same because they're saved in a separate database file in your site's folder. That said, pay attention to any extra post config options and templates that are unique to your old theme.

![Publii_Post options](https://github.com/user-attachments/assets/87278e34-6a55-49de-97be-5b01d01206e9)

Lastly: any [theme overrides](https://getpublii.com/dev/theme-overrides/) that you've done (if you're a pro theme customizer) won't be reflected in your new theme (obviously, but still worth mentioning).

**Reply by jjets718** [reply]
Very helpful! I successfully migrated themes a bit earlier today.
